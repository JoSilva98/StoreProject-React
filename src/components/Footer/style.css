.footer {
    display: flex;
    justify-content: space-between;
    padding: 30px 40px;
    background-color: rgba(244, 235, 214, 0.6);
    font-family: 'Times New Roman', Times, serif;
    font-size: 15px;
    box-shadow: 0px 0px 10px #888888;
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

@media screen and (max-width: 360px) {
    .footer {
        flex-direction: column;
        align-items: center;
    }

    .footer_first_p {
        margin-bottom: 10px;   
    }
}

@media screen and (max-width: 275px) {
    .footer {
        font-size: 13px;
    }
}