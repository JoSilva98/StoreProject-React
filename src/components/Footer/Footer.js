import "./style.css"

export default function Footer() {
    return (
        <footer className="footer" data-testid="footer-1">
            <p className="footer_first_p">João Silva</p>
            <p>© 2023. All Rights Reserved.</p>
        </footer>
    )
}