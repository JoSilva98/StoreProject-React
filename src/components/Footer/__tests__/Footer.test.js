import '@testing-library/jest-dom'
import { render, screen, cleanup } from "@testing-library/react"
import renderer from "react-test-renderer"
import Footer from "../Footer"

test("should render Footer", () => {
    render(<Footer />)

    // Vai buscar a componente renderizada com base no id
    const footerElement = screen.getByTestId("footer-1")

    expect(footerElement).toBeInTheDocument()
    expect(footerElement).toHaveTextContent("João Silva© 2023. All Rights Reserved.")
    expect(footerElement).not.toHaveTextContent("Hello World!")
})

test("Footer should match snapshot", () => {
    const tree = renderer.create(<Footer />).toJSON()
    expect(tree).toMatchSnapshot()
})