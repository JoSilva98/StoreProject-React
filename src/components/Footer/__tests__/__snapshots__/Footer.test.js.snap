// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<footer
  className="footer"
  data-testid="footer-1"
>
  <p
    className="footer_first_p"
  >
    João Silva
  </p>
  <p>
    © 2023. All Rights Reserved.
  </p>
</footer>
`;
